app/components/SessionNeedsReviewWarningComponent.rb

```diff
@@ -0,0 +1,30 @@
+class SessionNeedsReviewWarningComponent < ViewComponent::Base
+  erb_template <<-ERB
+    <div class="nhsuk-warning-callout">
+      <h3 class="nhsuk-warning-callout__label">
+        <span role="text">
+          <span class="nhsuk-u-visually-hidden">Important: </span>
+          Needs review
+        </span>
+      </h3>
+
+      <ul>
+        <% if @session.patients.without_nhs_number.count > 0 %>
+          <li>
+            <%= link_to t(:children_without_nhs_number, count: @session.patients.without_nhs_number.count),
+                        session_consent_path(@session, missing_nhs_number: true) %>
+          </li>
+        <% end %>
+      </ul>
+    </div>
+  ERB
+
+  def initialize(session:)
+    super
+    @session = session
+  end
+
+  def render?
+    @session.patients.without_nhs_number.exists?
```

Comment:
Similar to my other comment, we need to make sure we're counting these correctly and excluding patients who are in the session because they go to the school, but not in the right year group for any of the programmes.